File: 1513B.go
package main

import (
	"bufio"
	. "fmt"
	"os"
)

var mod int = 1e9 + 7

func main() {
	var t int
	in := bufio.NewReader(os.Stdin)
	for Fscan(in, &t); t > 0; t-- {
		var n int
		Fscan(in, &n)
		a := make([]int, n)
		x := -1
		for i := range a {
			Fscan(in, &a[i])
			x &= a[i]
		}
		cnt := 0
		for _, num := range a {
			if num == x {
				cnt++
			}
		}
		ans := cnt * (cnt - 1) % mod
		for i := 2; i <= n-2; i++ {
			ans = ans * i % mod
		}
		Println(ans)
	}
	return
}

File: 1513B.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
#include <utility>
#include <map>
using ll = long long;
using namespace std;

const ll mod = 1e9 + 7;

vector<ll> f(200005, 0);

ll factorial(ll x) {
	if (x == 0 || x == 1)
		return f[x] = 1;
	if (f[x] > 0)
		return f[x];
	return f[x] = x * factorial(x - 1) % mod;
}

int main() {
	std::ios::sync_with_stdio(false);
	cin.tie(nullptr);
	int t;
	cin >> t;
	while (t--) {
		int n;
		cin >> n;
		vector<int> a(n);
		int x = -1;
		for (int i = 0; i < n; ++i) {
			cin >> a[i];
			x &= a[i];
		}
		ll cnt = 0;
		for (int num : a) {
			if (num == x)
				++cnt;
		}
		cout << cnt * (cnt - 1) % mod * factorial(n - 2) % mod << endl;
		/*ll ans = cnt * (cnt - 1) % mod;
		for (int i = 2; i <= n - 2; i++) {
			ans = ans * i % mod;
		}
		cout << ans << endl;*/
	}

	return 0;

}